lib/pbio/drv/uart/uart_debug_first_port.c

```diff
@@ -33,14 +33,14 @@ static pbdrv_uart_dev_t *debug_uart = NULL;
  * @param va_list The variable arguments, as a va_list.
  */
 void pbdrv_uart_debug_vprintf(const char *format, va_list args) {
-    char buf[BUF_SIZE];
-    vsnprintf(buf, sizeof(buf), format, args);
-    size_t len = strlen(buf);
-    for (size_t i = 0; i < len; i++) {
-        ring_buf[ring_head] = buf[i];
-        ring_head = (ring_head + 1) % BUF_SIZE;
+    if (!lwrb_is_ready(&ring_buffer)) {
+        lwrb_init(&ring_buffer, ring_buf_storage, sizeof(ring_buf_storage));
     }
 
+    char buf[256];
```

Comment:
Since a typical terminal is 80 characters wide, I would be tempted to make this even smaller (e.g. 96). We can always do 2 prints if we _really_ want to have long lines.